app/controllers/account_activations_controller.rb

```diff
@@ -0,0 +1,12 @@
+class AccountActivationsController < ApplicationController
+  def edit
+    user = User.find_by(email: params[:email])
+    if user && !user.activated? && user.authenticated?(:activation, params[:id])
```

Comment:
This line violates soo many SRP principles.

Reply:
Ok, I have fixed the method.